src/testing/models.ts

```diff
@@ -20,21 +20,59 @@ export interface TracerEvent {
   properties: ArbitraryProperties;
 }
 
-export abstract class BaseTestEvaluator<TestCaseType, OutputType> {
+interface TestEvaluator<TestCaseType, OutputType> {
+  evaluateTestCase(args: {
+    testCase: TestCaseType;
+    output: OutputType;
+  }): Evaluation | Promise<Evaluation>;
+}
+
+interface EventEvaluator {
+  evaluateEvent(args: { event: TracerEvent }): Evaluation | Promise<Evaluation>;
+}
+
+abstract class _BaseEvaluator {
   abstract get id(): string;
 
   maxConcurrency: number = 10;
+}
 
+/**
+ * An ABC for users that are implementing an evaluator that will only be run against test cases.
+ */
+export abstract class BaseTestEvaluator<TestCaseType, OutputType>
+  extends _BaseEvaluator
+  implements TestEvaluator<TestCaseType, OutputType>
+{
   abstract evaluateTestCase(args: {
     testCase: TestCaseType;
     output: OutputType;
   }): Evaluation | Promise<Evaluation>;
 }
 
-export abstract class BaseEventEvaluator {
-  abstract get id(): string;
+/**
+ * An ABC for users that are implementing an evaluator that will only be run against production events.
+ */
+export abstract class BaseEventEvaluator
+  extends _BaseEvaluator
+  implements EventEvaluator
+{
+  abstract evaluateEvent(args: {
+    event: TracerEvent;
+  }): Evaluation | Promise<Evaluation>;
+}
 
-  maxConcurrency: number = 10;
+/**
+ * An ABC for users that are implementing an evaluator that will be run against both test cases and production events.
+ */
+export abstract class BaseEvaluator<TestCaseType, OutputType>
```

Comment:
@dmorton2297 @adamnolte lmk what you think of this name

Reply:
We should move this out of testing and export it from both testing and tracer packages

Reply:
hmm we discussed this already and decided to keep everything in testing

Reply:
It's pretty awkward to have to import the testing package into my main app when I'm just using the tracer

Reply:
It's not a huge deal but it did take me a little bit to figure out how to import this in one of my demo apps where I was not using testing.

Reply:
where the discussion landed previously is that "testing" is relevant in both online and offline scenarios so i don'tt think it's weird to import from testing

Reply:
ok i'm open to changing we just need to keep these models together IMO, so maybe we can have a new thing called `evaluators` just for these types/models and we can import from there in both online and offline scenarios

Reply:
Ya I think having an evaluators package makes sense.

Reply:
what do you think of the name though? wasn't sure if i should try to make it more clear that it's both in the name

Reply:
I think the name makes sense